msal/src/Microsoft.Identity.Client/Platforms/uap/PlatformInformation.cs

```diff
@@ -49,6 +49,71 @@ public override string GetProductName()
             return "MSAL.UAP";
         }
 
+        /// <summary>
+        /// Get a principal name that can be used for WIA
+        /// </summary>
+        /// <remarks>
+        /// Win10 allows several identities to be logged in at once; 
+        /// select the first principal name that can be used
+        /// </remarks>
+        /// <returns></returns>
+        public override async Task<string> GetUserPrincipalNameAsync()
+        {
+            IReadOnlyList<User> users = await User.FindAllAsync();
+            if (users == null || !users.Any())
+            {
+                throw new MsalClientException(
+                 CoreErrorCodes.CannotAccessUserInformationOrUserNotDomainJoined,
+                 CoreErrorMessages.UapCannotFindDomainUser);
```

Comment:
nit: should the names of the code and the message be the same? 
aside: we could have a class which held both the code and the message...

Reply:
Will use the same error message, good point. Will look at 1 class for code + message later.

---
In reply to: [220986182](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1276#discussion_r220986182) [](ancestors = 220986182)